rust/src/basis.rs

```diff
@@ -106,7 +106,12 @@ impl Basis {
         qweight1d: &[f64],
     ) -> Self {
         let mut ptr = std::ptr::null_mut();
-        let (dim, ncomp, P1d, Q1d) = (dim as i32, ncomp as i32, P1d as i32, Q1d as i32);
+        let (dim, ncomp, P1d, Q1d) = (
+            i32::try_from(dim).unwrap(),
+            i32::try_from(ncomp).unwrap(),
+            i32::try_from(P1d).unwrap(),
+            i32::try_from(Q1d).unwrap(),
+        );
```

Comment:
These unwraps will be eventually folded into our future error handling strategy w/Results

Reply:
Are we sure we want to use `unwrap` instead of `?`?

Reply:
? only works in functions that return Results

Reply:
Okay, that makes sense, thank you.